pattoo_shared/phttp.py

```diff
@@ -252,6 +432,198 @@ def post(url, data, identifier, save=True):
     return success
 
 
+def key_exchange(gpg, req_session, exchange_url, validation_url,
```

Comment:
Where's the unittest for this?

Reply:
This test is incorporated with the EncryptedPost unit test.